controller/class.tapestry-controller.php

```diff
@@ -43,54 +43,86 @@ class TapestryController
         'INVALID_CHILD_NODE' => [
             'MESSAGE'   => 'Node is not a child of the tapestry',
             'STATUS'    => ['status' => 400]
+        ],
+        'INVALID_NEW_TAPESTRY' => [
+            'MESSAGE'   => 'Tapestry should have at max one root node when being created',
+            'STATUS'    => ['status' => 400]
+        ],
+        'INVALID_NEW_LINK' => [
+            'MESSAGE'   => 'New Tapestry link is invalid',
+            'STATUS'    => ['status' => 400]
+        ],
+        'POST_ID_ALREADY_SET' => [
+            'MESSAGE'   => 'PostID is set when creating a new Tapestry',
+            'STATUS'    => ['status' => 500]
         ]
     ];
     private $postId;
 
     /**
      * Constructor
      */
-    public function __construct($postId = 0)
+    public function __construct($postId)
     {
-        if ($postId && !$this->_isValidTapestry($postId)) {
-            return $this->_throwsError('INVALID_POST_ID');
+        if ($postId && $this->_isValidTapestry($postId)) {
+            $this->postId = (int)$postId;
+        } else {
+            $this->postId = null;
         }
-        $this->postId = (int)$postId;
     }
 
     /**
-     * Update Tapestry nodes first then
-     * Update the existing Tapestry if the postId is provided
-     * Otherwise, a new Tapestry will be created
+     * Add A Tapestry
+     * 
+     * Note: the Tapestry should have at max one root node
+     * There should not be any links available at this moment
      * 
      * @param   Object  $tapestry   Tapestry
      * 
      * @return  Object  $tapestry
      */
-    public function updateTapestry($tapestry)
+    public function addTapestry($tapestry)
     {
-        if (!$this->postId) {
-            $this->postId = $this->_updatePost($tapestry, 'tapestry');
+        if ($this->postId) {
+            return $this->_throwsError('POST_ID_ALREADY_SET');
+        }
+        if (isset($tapestry->links) || count($tapestry->nodes) > 1) {
+            return $this->_throwsError('INVALID_NEW_TAPESTRY');
         }
         if (!isset($tapestry->nodes)) {
             $tapestry->nodes = [];
         }
         if (!isset($tapestry->groups)) {
             $tapestry->groups = [];
         }
+
+        $this->postId = $this->_updatePost($tapestry, 'tapestry');
+
```

Comment:
I suggest you dedicate this function solely to adding a new tapestry, not its nodes, groups, etc.